tests/unit/amo/components/TestConfirmButton.js

```diff
@@ -27,138 +22,147 @@ describe(__filename, () => {
-    const children = 'Do you really want to delete this?';
-    const root = render({ children });
+    render({ children: 'Do you really want to delete this?' });
 
-    expect(root.find(Button).children()).toHaveText(children);
+    const button = screen.getByRole('button', {
+      name: 'Do you really want to delete this?',
+    });
+    expect(button).toHaveClass('ConfirmButton-default-button');
+    expect(button).toHaveClass('Button--neutral');
   });
 
   it('shows ConfirmationDialog when button is clicked', () => {
-    const { store } = dispatchClientMetadata();
-    const root = render({ store });
+    render();
+    const button = screen.getByRole('button', {
+      name: 'the default text of this button',
+    });
 
-    expect(root).not.toHaveClassName('ConfirmButton--show-confirmation');
-    expect(root.find(ConfirmationDialog)).toHaveLength(0);
+    expect(
+      screen.queryByClassName('ConfirmButton--show-confirmation'),
+    ).not.toBeInTheDocument();
+    expect(screen.queryByText('some warning message')).not.toBeInTheDocument();
 
-    root.find(Button).simulate('click', createFakeEvent());
-    applyUIStateChanges({ root, store });
+    userEvent.click(button);
 
-    expect(root).toHaveClassName('ConfirmButton--show-confirmation');
-    expect(root.find(ConfirmationDialog)).toHaveLength(1);
+    expect(screen.getByText('some warning message')).toBeInTheDocument();
+    expect(screen.getByClassName('ConfirmationDialog')).toBeInTheDocument();
+    expect(
+      screen.getByClassName('ConfirmButton--show-confirmation'),
+    ).toBeInTheDocument();
   });
 
   it('configures ConfirmationDialog', () => {
     const cancelButtonText = 'Nevermind, take me back';
-    const cancelButtonType = 'alert';
     const confirmButtonText = 'Do it!';
-    const confirmButtonType = 'alert';
     const message = 'Do you really want to cancel?';
-    const puffyButtons = true;
 
-    const root = renderWithDialog({
+    renderWithDialog({
       cancelButtonText,
-      cancelButtonType,
+      cancelButtonType: 'alert',
       confirmButtonText,
-      confirmButtonType,
+      confirmButtonType: 'alert',
       message,
-      puffyButtons,
+      puffyButtons: true,
     });
 
-    const confirmDialog = root.find(ConfirmationDialog);
-    expect(confirmDialog).toHaveProp('cancelButtonText', cancelButtonText);
-    expect(confirmDialog).toHaveProp('cancelButtonType', cancelButtonType);
-    expect(confirmDialog).toHaveProp('confirmButtonText', confirmButtonText);
-    expect(confirmDialog).toHaveProp('confirmButtonType', confirmButtonType);
-    expect(confirmDialog).toHaveProp('message', message);
-    expect(confirmDialog).toHaveProp('puffyButtons', puffyButtons);
+    expect(screen.getByText(message)).toBeInTheDocument();
+
+    const dialogCancelButton = screen.getByRole('button', {
+      name: cancelButtonText,
+    });
+    expect(dialogCancelButton).toHaveClass('ConfirmationDialog-cancel-button');
```

Comment:
I don't think we need to assert about these class names, especially as they are not even part of the code in `ConfirmButton`. They are in the code for `ConfirmationDialog` and therefore would be covered in tests for `ConfirmationDialog`.